### PR TITLE
docs: terminology doc

### PR DESCRIPTION
**Pull request by vinckr**
Documenting terms that are often used in Ory documentation and Ory Cloud specific terms.

### CODE REVIEW

File: docs/docs/concepts/terminology.mdx [hunk 1]
```diff
@@ -5,55 +5,85 @@ title: Terminology
...
+
+#### API Client:
+
+End-user side. A browser client refers to the Web browser in the user's machine
+or mobile device.
 
 ## Flows
 
-#### Registration Flow: 
+#### Registration Flow:
```
Comment (aeneasr) [inline docs/docs/concepts/terminology.mdx:43]: Remove the dots after headlines? :)

Comment (vinckr) [reply docs/docs/concepts/terminology.mdx]: good point 👍

**Comment by vinckr** [comment]
Added some explanation for these terms in https://github.com/ory/docs/pull/468/commits/78030b4a43d4dac3d91968b5ac0a23386b2ebe6c .

**Comment by vinckr** [comment]
I think this is good to go, let me know if there is any terms I missed or if you have feedback on the layout.

**Review by aeneasr — APPROVED** [review]

**Pull request merged**